Ginger/GingerTest/ActionConversion/ActionConversionTest.cs

```diff
@@ -0,0 +1,331 @@
+using System.IO;
+using System.Linq;
+using GingerCore.Actions;
+using Amdocs.Ginger.CoreNET;
+using System.Collections.Generic;
+using GingerCore.Actions.Common;
+
+namespace GingerTest
+{
+    [TestClass]
+    [Level1]
+    public class ActionConversionTest
+    {
+        static SolutionRepository mSolutionRepository;
+        static BusinessFlow mBF;
+        static string solutionName; 
+
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext TC)
+        {
+            solutionName = "ActionConversionSol";
+            CreateTestSolution();
+
+            // Creating workspace
+            WorkSpace.Init(new WorkSpaceEventHandler());
+            WorkSpace.Instance.SolutionRepository = GingerSolutionRepository.CreateGingerSolutionRepository();
+
+            // Init SR
+            mSolutionRepository = WorkSpace.Instance.SolutionRepository;
+            Ginger.App.InitClassTypesDictionary();
+            string TempRepositoryFolder = TestResources.GetTestTempFolder(@"Solutions\" + solutionName);
+            mSolutionRepository.Open(TempRepositoryFolder);
+        }
+
+        [TestCleanup]
+        public void TestCleanUp()
+        {
+
+        }
+
+        private static void CreateTestSolution()
+        {
+            // First we create a basic solution with some sample items
+            SolutionRepository SR = new SolutionRepository();
+            string TempRepositoryFolder = TestResources.GetTestTempFolder(@"Solutions\" + solutionName);
+            if (Directory.Exists(TempRepositoryFolder))
+            {
+                Directory.Delete(TempRepositoryFolder, true);
+            }
+
+            SR = GingerSolutionRepository.CreateGingerSolutionRepository();
+            SR.Open(TempRepositoryFolder);
+
+            SR.Close();
+        }
+        
+        [TestMethod]
+        [Timeout(60000)]
+        public void ActionConversionToSameActivityTest()
+        {
+            mBF = new BusinessFlow() { Name= "TestBFConversion", Active=true };
+            
+            Activity activity = new Activity();
+            activity.SelectedForConversion = true;
+            ActGenElement gen1 = new ActGenElement();
+            gen1.Active = true;
+            gen1.Description = "Set Value : first_name input";
+            gen1.LocateBy = Amdocs.Ginger.Common.UIElement.eLocateBy.ByRelXPath;
+            gen1.LocateValue = "//input[@name='first_name']";
+            gen1.GenElementAction = ActGenElement.eGenElementAction.SendKeys;
+            activity.Acts.Add(gen1);
+
+            ActGenElement gen2 = new ActGenElement();
+            gen2.Active = true;
+            gen2.Description = "Set Value : last_name input";
+            gen2.LocateBy = Amdocs.Ginger.Common.UIElement.eLocateBy.ByName;
+            gen2.LocateValue = "last_name";
+            gen2.GenElementAction = ActGenElement.eGenElementAction.SendKeys;
+            activity.Acts.Add(gen2);
+
+            mBF.AddActivity(activity);
```

Comment:
codes in lines 113-124 is same in all of your test methodes- please pull it into function which will be called from all of those methodes

Reply:
Done